routes/mentors.js

```diff
@@ -99,4 +99,48 @@ router.patch("/mentors/:id", verifyRoles([ROLE_NAMES.member]), async (req, res)
   }
 });
 
+// Update mentor information by user owning mentorship card
+router.put("/mentors/:id", verifyRoles([ROLE_NAMES.mentor]), async (req, res) => {
+  const { id } = req.params;
+  const { updates } = req.body;
+  
+  if (!isValidIntegerId(id)) {
+    return respondWithError(res, 400, "Invalid user id supplied");
+  }
+
+  // check that only fields 'about' and 'contact' can be updated
+  const invalidFields = Object.keys(updates)
+    .filter(field => !ALLOWED_MENTOR_UPDATES.includes(field));
+  if(invalidFields.length) {
+    return respondWithError(res, 400, "You are not allowed to edit this field");
```

Comment:
I think 400 is rather you _cant_ edit instead of not _allowed_

Reply:
also 'this field' is not very clear. should we return the names of the fields from the payload that we didn't accept?